### PR TITLE
Added empty namespaceConfusionAttackPreventionJob job CLM-22306

### PR DESCRIPTION
**Pull request by isolonyi-sonatype**
#### Description
Added new job
#### Links
JIRA: [CLM-22306](https://issues.sonatype.org/browse/CLM-22306)
Build: [CLM-22306]( https://jenkins.ci.sonatype.dev/job/insight/job/firewall-for-artifactory/job/feature-snapshots/job/CLM-22306_add_new_job/)

### CODE REVIEW

**Comment by vladt** [comment]
Please use a more meaningful title for this PR

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy [hunk 1]
```diff
@@ -66,6 +66,8 @@ class FirewallProperties
 
   public static final String FIREWALL_VERIFY_WAIT_MILLIS = 'firewall.verify.wait.in.millis'
 
+  public static final String JOB_ENABLE = 'job.enable'
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy:69]: We need to use a much more specific property name here

File: firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy [hunk 2]
```diff
@@ -41,6 +41,10 @@ trait NexusFirewallForArtifactory
 
   abstract void auditRepositories()
 
+  abstract def getNamespaceConfusionJobCronExpression()
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy:44]: def->String?

**Comment by vladt** [comment]
Please use a more meaningful title for this PR

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy (hunk 1 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy:69]: deleted as we using cron specific approach to enable/disable jobs

File: firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy (hunk 2 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy:44]: changed

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy [hunk 3]
```diff
@@ -147,6 +148,10 @@ class FirewallProperties
     return properties[FIREWALL_CACHE_EXPIRE_AFTER_WRITE_IN_MILLIS] as Long
   }
 
+  String getJobEnabler() {
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy:151]: Is this needed? Especially since you removed JOB_ENABLE?

Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy:151]: you are right, we don't need it for now

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy [hunk 4]
```diff
@@ -66,6 +66,7 @@ class FirewallProperties
 
   public static final String FIREWALL_VERIFY_WAIT_MILLIS = 'firewall.verify.wait.in.millis'
 
+
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy:69]: Unnecessary change

File: firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy [hunk 5]
```diff
@@ -219,6 +221,11 @@ class DefaultNexusFirewallForArtifactory
     }
   }
 
+  @Override
+  void namespaceConfusionAttackPreventionJobHandler() {
+    log.info("test handler");
```
Comment (fernau) [inline firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy:226]: accourding to the A/C:
> should wait for the plugin to be fully initialized before doing anything at all
there are a few examples in the existing code that already do that. you just need do to the same thing here.
please also add a test.

**Review by DanielRuaMadrid — APPROVED** [review]

File: firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy [hunk 6]
```diff
@@ -41,6 +41,10 @@ trait NexusFirewallForArtifactory
 
   abstract void auditRepositories()
 
+  abstract String getNamespaceConfusionJobCronExpression()
```
Comment (vladt) [inline firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy:44]: These two methods refer to the same feature, so they should have consistent naming. So, either we use the long form (namespaceConfusionAttackPrevention) or the short form (namespaceConfusion) everywhere, but not both.

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy (hunk 1 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy]: yes

File: plugin/src/main/java/com/sonatype/iq/artifactory/nexusFirewallForArtifactoryPlugin.groovy [hunk 7]
```diff
@@ -135,6 +135,34 @@ storage {
...
+ * The actual code to run as part of the job should be part of the job's closure.
+ *
+ * Parameters:
+ * delay (long) - An initial delay in milliseconds before the job starts running (not applicable for a cron job).
+ * interval (long) -  An interval in milliseconds between job runs.
+ * cron (java.lang.String) - A valid cron expression used to schedule job runs
+ * **/
+jobs {
+  namespaceConfusionAttackPreventionJob(cron: this.nexusFirewallForArtifactory.getNamespaceConfusionJobCronExpression())
+      {
```
[the remark this answers is not in the document]
Comment (isolonyi-sonatype) [reply plugin/src/main/java/com/sonatype/iq/artifactory/nexusFirewallForArtifactoryPlugin.groovy:152]: https://github.com/sonatype/firewall-for-jfrog-artifactory/pull/11/commits/91d0f496d098d2e7ffda80e2c9adbd77b7e9477e formatted

File: firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy (hunk 6 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy]: https://github.com/sonatype/firewall-for-jfrog-artifactory/pull/11/commits/91d0f496d098d2e7ffda80e2c9adbd77b7e9477e now we are using long names

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy (hunk 4 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy]: deleted

File: firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy [hunk 8]
```diff
@@ -219,6 +221,14 @@ class DefaultNexusFirewallForArtifactory
     }
   }
 
+  @Override
+  void namespaceConfusionAttackPreventionJobHandler() {
+    if (initializationVerified.getCount() != 0) {
+      throw new CancelException('Invoked namespaceConfusionAttackPreventionJobHandler before initialisation.', 500)
```
[the remark this answers is not in the document]
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy]: changed, now without Cancelation exception

File: firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy (hunk 5 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/DefaultNexusFirewallForArtifactory.groovy]: added

File: plugin/src/main/java/com/sonatype/iq/artifactory/nexusFirewallForArtifactoryPlugin.groovy (hunk 7 above)
[the remark this answers is not in the document]
Comment (isolonyi-sonatype) [reply plugin/src/main/java/com/sonatype/iq/artifactory/nexusFirewallForArtifactoryPlugin.groovy:152]: > Formatting
https://github.com/sonatype/firewall-for-jfrog-artifactory/commit/91d0f496d098d2e7ffda80e2c9adbd77b7e9477e formatted

File: firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy (hunk 6 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/NexusFirewallForArtifactory.groovy]: > These two methods refer to the same feature, so they should have consistent naming. So, either we use the long form (namespaceConfusionAttackPrevention) or the short form (namespaceConfusion) everywhere, but not both.
https://github.com/sonatype/firewall-for-jfrog-artifactory/commit/91d0f496d098d2e7ffda80e2c9adbd77b7e9477e now we are using long names

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy (hunk 4 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy]: > Unnecessary change

deleted

File: firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy (hunk 3 above)
Comment (isolonyi-sonatype) [reply firewall/src/main/java/com/sonatype/iq/artifactory/FirewallProperties.groovy]: > Is this needed? Especially since you removed JOB_ENABLE?
you are right, we don't need it for now
